### PR TITLE
Add latch

### PR DESCRIPTION
**Pull request by nh13**
Adds the [`latch` CLI](https://github.com/latchbio/latch) by [Latch Bio](https://latch.bio/). Attempts to do so by adding all custom `lyte*` sub-packages in one recipe.

Alternative, is to have each package in its own recipe here: #40953, #40954, #40955, and #40956

Status: tests are currently failing, and I have no clue how to fix.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by nh13** [comment]
@dpryan79 thanks for your help, I was able to get this working locally if I run `conda build --no-test .`.  I am still unsure why the tests fail when the same tests pass here: https://github.com/bioconda/bioconda-recipes/pull/40953.  Perhaps it's something to do with the install name in the `script: ... pip install` command?

**Comment by dpryan79** [comment]
I think we've made some progress finally.

File: recipes/latch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+build:
+  noarch: python
+  number: 0
+
+package:
+  name: {{ name|lower }}
+  version: {{ latch_version }}
+
+requirements:
```
Comment (nh13) [inline recipes/latch/meta.yaml:15]:
```suggestion
requirements:
  host:
    - python >=3.8
```

**Comment by dpryan79** [comment]
Azure is being problematic, hopefully this will be resolved soon.

**Comment by nh13** [comment]
@BiocondaBot please fetch artifacts

**Comment by nh13** [comment]
@BiocondaBot please fetch artifacts

**Comment by nh13** [comment]
@dpryan79 do you have any idea why I can't fetch artifacts?  I looked at the linux tests and see the following in the "prepare" and "archive packages" steps (why is `krakenuniq` there?).

```
Run (
'index.html' -> '/tmp/artifacts/packages/index.html'
'channeldata.json' -> '/tmp/artifacts/packages/channeldata.json'
'linux-64' -> '/tmp/artifacts/packages/linux-64'
'linux-64/index.html' -> '/tmp/artifacts/packages/linux-64/index.html'
'linux-64/repodata.json' -> '/tmp/artifacts/packages/linux-64/repodata.json'
'linux-64/repodata.json.bz2' -> '/tmp/artifacts/packages/linux-64/repodata.json.bz2'
'linux-64/current_repodata.json' -> '/tmp/artifacts/packages/linux-64/current_repodata.json'
'linux-64/current_repodata.json.bz2' -> '/tmp/artifacts/packages/linux-64/current_repodata.json.bz2'
'linux-64/krakenuniq-1.0.4-pl5321h19e8d03_0.tar.bz2' -> '/tmp/artifacts/packages/linux-64/krakenuniq-1.0.4-pl5321h19e8d03_0.tar.bz2'
'linux-64/repodata_from_packages.json.bz2' -> '/tmp/artifacts/packages/linux-64/repodata_from_packages.json.bz2'
'linux-64/repodata_from_packages.json' -> '/tmp/artifacts/packages/linux-64/repodata_from_packages.json'
cp: cannot stat 'osx-64': No such file or directory
'noarch' -> '/tmp/artifacts/packages/noarch'
'noarch/repodata.json.bz2' -> '/tmp/artifacts/packages/noarch/repodata.json.bz2'
'noarch/current_repodata.json' -> '/tmp/artifacts/packages/noarch/current_repodata.json'
'noarch/current_repodata.json.bz2' -> '/tmp/artifacts/packages/noarch/current_repodata.json.bz2'
'noarch/repodata.json' -> '/tmp/artifacts/packages/noarch/repodata.json'
'noarch/repodata_from_packages.json.bz2' -> '/tmp/artifacts/packages/noarch/repodata_from_packages.json.bz2'
'noarch/index.html' -> '/tmp/artifacts/packages/noarch/index.html'
'noarch/latch-2.19.11-pyhdfd78af_0.tar.bz2' -> '/tmp/artifacts/packages/noarch/latch-2.19.11-pyhdfd78af_0.tar.bz2'
'noarch/repodata_from_packages.json' -> '/tmp/artifacts/packages/noarch/repodata_from_packages.json'
Start compressing docker image quay.io/biocontainers/latch:2.19.11--pyhdfd78af_0 ...
Finished compressing docker image quay.io/biocontainers/latch:2.19.11--pyhdfd78af_0 .
1s
```

```
Run actions/upload-artifact@v3
With the provided path, there will be 18 files uploaded
Starting artifact upload
For more detailed logs during the artifact upload process, enable step-debugging: https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging
Artifact name is valid!
Container for artifact "linux-packages" successfully created. Starting upload of file(s)
Total size of all the files uploaded is 9096[28](https://github.com/bioconda/bioconda-recipes/actions/runs/5070802734/jobs/9106400433#step:9:29) bytes
File upload process has finished. Finalizing the artifact upload
Artifact has been finalized. All files have been successfully uploaded!

The raw size of all the files that were specified for upload is 922095 bytes
The size of all the files that were uploaded is 90962[8](https://github.com/bioconda/bioconda-recipes/actions/runs/5070802734/jobs/9106400433#step:10:9) bytes. This takes into account any gzip compression used to reduce the upload size, time and storage

Note: The size of downloaded zips can differ significantly from the reported size. For more information see: https://github.com/actions/upload-artifact#zipped-artifact-downloads 

Artifact linux-packages has been successfully uploaded!
```

**Comment by nh13** [comment]
@BiocondaBot please add label

**Comment by nh13** [comment]
Tested installing the `linux-packages.zip` locally without any issues

**Comment by dpryan79** [comment]
Since we had to do an emergency switch to a different CI provider there's currently no way for the bot to fetch artifacts for you. To find them, migrate over to the github action for your commit and then the summary section. The artifacts are at the bottom.

**Review by dpryan79 — APPROVED** [review]

**Pull request merged**